frontend/cypress/integration/invite_spec.js

```diff
@@ -0,0 +1,114 @@
+describe('Invitation tests', () => {
+  before(() => {
+    cy.login() 
+
+    const fields = [
+      { label: 'Name', value: 'test new club', pressEnter: false },
+      { label: 'Size', value: '< 20', pressEnter: true },
+      {
+        label: 'Is an application required to join your organization?',
+        value: 'No Application Required',
+        pressEnter: true,
+      },
+    ]
+    
+    // Create club
+    cy.visit('/create')
+
+    fields.forEach(({ label, value, pressEnter }) => {
+      const input = cy.contains('.field', label).find('input')
+      input.focus().clear().type(value)
+
+      if (pressEnter) {
+        input.type('{enter}')
+      }
+
+      input.blur()
+    })
+
+    cy.contains('Select tags relevant to your club!').click()
+    cy.contains('Undergraduate').click()
+
+    cy.contains('Submit').click()
+
+    // wait for club to be created, should be redirected to renewal page
+    cy.contains('Renew Club').should('be.visible') 
```

Comment:
Got it. I am now using PPPJO as the club and the James Madison (jmadison@seas.upenn.edu) as the account Benjamin Franklin is inviting. I managed to find out why this was working on my local development while it wasn't on Circle CI.

Since I need to test the process of both inviting and accepting an invitation, I needed to use an account that has an email address. This prevents me from using the test user as the account being invited as it does not have a registered email address. 

While it would have been ideal to use the bfranklin account to invite himself to one of the clubs, unfortunately bfranklin seems to be registered for all the clubs. Hence, I chose to have bfranklin login and then invite jmadison (who is not an originally registered member of PPPJO). For jmadison to accept this invitation, I would have cypress login as jmadison. Unfortunately, inorder to login as jmadison, I would have to access the `api/admin/login/` route. This would be prevented on cypress as jamdison is **not** a staff.

The reason why it was working on my development environment is because I mistakenly ran the command `User.objects.all().update(is_superuser=True, is_staff=True)` making jmadison a staff. 

I managed to bypass this by starting off the testing suite in logging in as bfranklin and promoting jmadison as both a staff and super user.
https://github.com/pennlabs/penn-clubs/blob/036a2a93671f807ef1d108a15318209a0fcca47f/frontend/cypress/integration/invite_spec.js#L3-L11


I completely agree with you if you think this isn't the cleanest implementation as the process of promoting and demoting jmadison to become a staff/superuser ideally shouldn't be part of the testing suite. I think the more cleaner way to fix this is to have

Reply:
This is a very clever solution! This is completely acceptable for me.

The reason the test user isn't set up with an email is because that's part of how Platform auth works, which is outside of this codebase. We could potentially add a test email in the future if it is useful.